torch_xla/csrc/tensor.cpp

```diff
@@ -699,6 +709,14 @@ void XLATensor::SetInPlaceIrValue(torch::lazy::Value ir_value) {
 }
 
 void XLATensor::AssignIrValue(torch::lazy::Value ir_value) const {
+  // Sharding annotation it not null, if xla_data() is sharded.
+  // TODO(yeounoh): Sharding annotation must be removed by explicit call to
+  // ClearSharding.
+  ShardingSpecPtr sharding = sharding_spec();
+  if (sharding != nullptr) {
```

Comment:
we need a test for this. For example when we deep copy a tensor with sharding, the result tensor should also have sharding. Something similar to https://github.com/pytorch/xla/blob/b3f79ccf1c2e85f76c737eeeab0e4ca512d180b4/test/test_operations.py#L1670 

@steventk-g can you add a test case?

Reply:
Done https://github.com/pytorch/xla/issues/4095

Reply:
Good point, @steventk-g let me handle this if you haven't already started.